user-guide/Advanced_Functionality/Databases/Supported_System_Data_Storage_Architectures/Resource_migration_to_elastic.md

```diff
@@ -0,0 +1,139 @@
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<ResourceManagerConfig>
+  ...
+  <ResourceSettings>
+    <ResourceStorageType>Elastic</ResourceStorageType>
+  </ResourceSettings>
+</ResourceManagerConfig>
+```
+
+## Limitations and differences with Xml storage
+
+### Allowed property names
+
+Since the properties of resources and resource pools are indexed in Elasticsearch after the migration, there are some restrictions that apply:
+
+- Property names may not start with character ``_``.
+- Property names may not contain characters ``.`` (period), ``#`` (hashtag), ``*`` (star), ``,`` (comma), ``"`` (double quote) or ``'`` (single quote).
+- Property names cannot be empty or contain only whitespace characters.
+- Property values cannot be ``null``.
+
+If ElasticSearch storage is enabled and a resource or resource pool with invalid properties is added or updated, the API will return a ``ResourceManagerErrorData`` in the ``TraceData``, with reason ``InvalidCharactersInPropertyNames``.
+
+### Field size
+
+Field names in Elasticsearch have a maximum length of 32766 bytes, which means any field of a resource or resource pool indexed in Elasticsearch only contain 32766 bytes. This limit is mostly important for string fields, which can contain 32766 characters of one byte (or 16383 characters of two bytes). When trying to save a resource or resource pool with a field that is too big, the API will return an ``UnknownError``. The ``SLResourceManager.txt`` log file will contain the actual exception, which will mention the field that could not be indexed in Elasticsearch.
+
+### Number of indices
+
+When a resource is indexed in elasticsearch, all its properties, capacities and capabilities will be indexed as well. This means each unique property name and unique capacity and capability id will become a field mapping in Elasticsearch. If the index contains a lot of such mappings, Elasticsearch starts having performance problems. During testing, no issues were found when using a typical amount of capacities, capabilities and property names, but slowdowns started to be noticeable when more than 300 unique field mappings were present.
+
+### Difference when deleting resources
+
+When using ``Xml`` as storage, it is not allowed to remove a resource while one of the agents in the cluster is unreachable, as this could cause syncing issues. When using Elasticsearch as storage, this is allowed.
+
+### Initial event on subscriptions
+
+When using ``Xml`` as storage, if a client subscribes on the ``ResourceManagerEventMessage``, he will receive an initial event with all resources and resource pools. When using Elasticsearch as storage these events will no longer be sent. This is to prevent sending a message to the client containing thousands of resources.
```

Comment:
If you're using xml as storage and subscribe on the ..., you will

Reply:
Assuming that "a client" is a client application, it should be "it" instead of "he".

Reply:
Done